core/src/main/java/io/questdb/Server.java

```diff
@@ -0,0 +1,210 @@
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb;
+
+import io.questdb.cairo.*;
+import io.questdb.cutlass.Services;
+import io.questdb.cutlass.text.TextImportJob;
+import io.questdb.cutlass.text.TextImportRequestJob;
+import io.questdb.griffin.DatabaseSnapshotAgent;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.FunctionFactoryCache;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.engine.groupby.vect.GroupByJob;
+import io.questdb.griffin.engine.table.LatestByAllIndexedJob;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.mp.WorkerPool;
+import io.questdb.mp.WorkerPoolManager;
+import io.questdb.std.*;
+
+import java.util.ServiceLoader;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class Server implements QuietCloseable {
+
+    private final PropServerConfiguration config;
+    private final Log log;
+    private final ObjList<QuietCloseable> toBeClosed = new ObjList<>();
+    private final AtomicBoolean hasStarted = new AtomicBoolean();
+    private final WorkerPoolManager workerPoolManager;
+
+    public Server(String... args) throws SqlException {
+        this(Bootstrap.withArgs(args));
+    }
+
+    public Server(Bootstrap bootstrap) throws SqlException {
+        this(bootstrap.getConfig(), bootstrap.getMetrics(), bootstrap.getLog());
+    }
+
+    public Server(PropServerConfiguration config, Metrics metrics, Log log) throws SqlException {
+        this.config = config;
+        this.log = log;
+
+        // create cairo engine
+        final CairoConfiguration cairoConfig = config.getCairoConfiguration();
+        final CairoEngine cairoEngine = new CairoEngine(cairoConfig, metrics);
+        toBeClosed.add(cairoEngine);
+
+        // setup shared worker pool, plus dedicated pools
+        final FunctionFactoryCache ffCache = new FunctionFactoryCache(
+                cairoConfig,
+                ServiceLoader.load(FunctionFactory.class, FunctionFactory.class.getClassLoader())
+        );
+        final WorkerPool sharedPool = WorkerPoolManager.createUnmanaged(config.getWorkerPoolConfiguration(), metrics);
+        sharedPool.configureAsShared(cairoEngine, ffCache, true, true, true);
```

Comment:
package "mp" should not be making references to `cairo`, `griffin` etc. Otherwise it makes spaghetti